src/CFamily/CompilationDatabase/CompilationDatabaseLocator.cs

```diff
@@ -0,0 +1,130 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using Newtonsoft.Json;
+using SonarLint.VisualStudio.Infrastructure.VS;
+using SonarLint.VisualStudio.Integration;
+
+namespace SonarLint.VisualStudio.CFamily.CompilationDatabase
+{
+    internal interface ICompilationDatabaseLocator
+    {
+        /// <summary>
+        /// Returns absolute path to the compilation database file of the currently active build configuration.
+        /// Returns null if the file was not found.
+        /// </summary>
+        string Locate();
+    }
+
+    [Export(typeof(ICompilationDatabaseLocator))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class CompilationDatabaseLocator : ICompilationDatabaseLocator
```

Comment:
This logic is specific to VS-CMake, isn't it? It would be better if the class name/namespace reflected that (the `CMakeSettings` class is also VS-CMake-specific, so the same point about the namespace would apply).

? `SonarLint.VisualStudio.CFamily.CMake` ?